components/brave_wallet/browser/swap_request_helper.cc

```diff
@@ -22,15 +22,18 @@ absl::optional<std::string> EncodeJupiterTransactionParams(
   DCHECK(params);
   base::Value::Dict tx_params;
 
-  // feeAccount is the ATA account for the output mint where the fee will be
-  // sent to.
   absl::optional<std::string> associated_token_account =
       SolanaKeyring::GetAssociatedTokenAccount(
           params->output_mint, brave_wallet::kSolanaFeeRecipient);
   if (!associated_token_account)
     return absl::nullopt;
```

Comment:
This should only be run when HasJupiterFeesForTokenMint returns true.

Reply:
This block of code is additionally returning `absl::nullopt` for invalid `output_mint`.

If we move it inside the`if` condition, we'll lose the ability to detect an invalid `outout_mint` if `HasJupiterFeesForTokenMint(params->output_mint)` is `false`. That seemed to me like a step-down from the current behaviour.

Reply:
hm okay, please add a comment why we still need to get associated token account, otherwise it's a bit unclear since associated_token_account won't be used when HasJupiterFeesForTokenMint(params->output_mint) is false.

Reply:
Added a comment.